project/tickets/models.py

```diff
@@ -0,0 +1,134 @@
+from django.db import models
+from django.contrib.auth.models import User
+from django.utils import timezone
+
+
+# ----- TICKET TYPE ----- #
+class TicketType(models.Model):
+    TICKET_TYPE_CHOICES = (
+        ("Bug", "Bug"),
+        ("Feature", "Feature")
+    )
+    ticket_type = models.CharField(
+        max_length=7,
+        unique=True,
+        choices=TICKET_TYPE_CHOICES)
+
+    class Meta:
+        verbose_name=("Ticket Type")
+        verbose_name_plural=("Ticket Types")
+
+    def __str__(self):
+        return self.ticket_type
+
+
+# ----- TICKET STATUS ----- #
+class TicketStatus(models.Model):
+    TICKET_STATUS_CHOICES = (
+        ("Open", "Open"),
+        ("In Progress", "In Progress"),
+        ("Closed", "Closed"),
+    )
+    ticket_status = models.CharField(
+        max_length=11,
+        unique=True,
+        choices=TICKET_STATUS_CHOICES)
+
+    class Meta:
+        verbose_name=("Ticket Status")
+        verbose_name_plural=("Ticket Status")
+
+    def __str__(self):
+        return self.ticket_status
+
+
+# ----- TICKETS ----- #
+class Ticket(models.Model):
+    """
+    Allow users to open Tickets (Bugs or Features).
+    Default status should be 'Open' with current timestamp.
+    """
+    # user fields (editable)
+    title = models.CharField(
+        max_length=75,
+        default="",
+        blank=False)
+    description = models.TextField(
+        max_length=2000,
+        default="",
+        blank=False)
+
+    # ticket fields (non-editable)
+    author = models.ForeignKey(
+        User,
+        null=True,
+        on_delete=models.CASCADE)
+    ticket_status = models.ForeignKey(
+        TicketStatus,
+        null=True)
+    ticket_type = models.ForeignKey(
+        TicketType,
+        null=True)
+    date_created = models.DateTimeField(
+        blank=False,
+        null=False,
+        auto_now_add=True)
+    date_edited = models.DateTimeField(
+        blank=False,
+        null=False,
+        default=timezone.now)
```

Comment:
use `auto_now` instead, so it enters the current time whenever the model is updated

Reply:
auto_now doesn't work for this field properly. I originally tried it like that, but wouldn't work. I tried again two different ways. _Option 1:_ auto_now=True updates **every single ticket** to show it was updated at that exact moment. _Option 2:_ default=auto_now gives me the following error: **NameError: name 'auto_now' is not defined** so I have reverted back to timezone.now in this instance